File: frontend/src/app/features/guias/guide-detail.ts
import { HttpErrorResponse } from '@angular/common/http';
import { ChangeDetectionStrategy, Component, inject, OnInit, signal } from '@angular/core';
import { ActivatedRoute, RouterLink } from '@angular/router';
import { TranslatePipe } from '@ngx-translate/core';
import { GuideDetail as GuideDetailModel, GuidesApi } from './guias.api';
import { formatBrDate, GUIDE_STATUS_BADGE, showsAuthPassword } from './guide-format';

/**
 * Guide detail (SPEC-0012 BR5/BR7): items table (TUSS code, description, quantity, item status);
 * when AUTORIZADA/PARCIALMENTE_AUTORIZADA, the authorization password + validity; an expired
 * password (`authExpired`) shows the dedicated notice (BR7); NEGADA shows the denial reason; plus
 * contact guidance. A missing/out-of-scope id renders a dedicated "não encontrada" state (404
 * `guide.not-found`), distinct from a transient/unexpected failure — mirrors DocumentDetail
 * (features/minha-saude/document-detail.ts).
 */
@Component({
  selector: 'app-guide-detail',
  imports: [RouterLink, TranslatePipe],
  templateUrl: './guide-detail.html',
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class GuideDetail implements OnInit {
  private readonly api = inject(GuidesApi);
  private readonly route = inject(ActivatedRoute);

  protected readonly loading = signal(true);
  protected readonly notFound = signal(false);
  protected readonly errorKey = signal<string | null>(null);
  protected readonly guide = signal<GuideDetailModel | null>(null);

  protected readonly badgeClass = GUIDE_STATUS_BADGE;
  protected readonly formatBrDate = formatBrDate;
  protected readonly showsAuthPassword = showsAuthPassword;

  ngOnInit(): void {
    this.load();
  }

  load(): void {
    const id = this.route.snapshot.paramMap.get('id');
    if (!id) {
      return;
    }
    this.loading.set(true);
    this.notFound.set(false);
    this.errorKey.set(null);
    this.api.getGuide(id).subscribe({
      next: (detail) => {
        this.guide.set(detail);
        this.loading.set(false);
      },
      error: (error: HttpErrorResponse) => {
        this.loading.set(false);
        if (error.error?.code === 'guide.not-found') {
          this.notFound.set(true);
        } else {
          this.errorKey.set('common.error');
        }
      },
    });
  }
}

File: backend/src/main/java/com/fkmed/domain/guides/GuideDetail.java
package com.fkmed.domain.guides;

import java.time.LocalDate;
import java.util.List;
import java.util.UUID;

/**
 * The guide detail (SPEC-0012 BR5): the common header plus its items, the authorization password
 * and validity (only when {@link #status} is {@link GuideStatus#AUTORIZADA} or {@link
 * GuideStatus#PARCIALMENTE_AUTORIZADA}), the derived {@link #authExpired} badge and the denial
 * reason (only when {@link #status} is {@link GuideStatus#NEGADA}).
 */
public record GuideDetail(
    UUID id,
    String number,
    GuideType type,
    String requestingProvider,
    LocalDate requestedAt,
    GuideStatus status,
    String authPassword,
    LocalDate authValidUntil,
    boolean authExpired,
    String denialReason,
    List<ItemView> items) {

  /** One item of the guide (BR5): TUSS code, description, quantity and its own status. */
  public record ItemView(
      String tussCode, String description, int quantity, GuideItemStatus status) {}
}
